replica/table.cc

```diff
@@ -3011,6 +3029,18 @@ table::mutation_query(schema_ptr s,
         co_return reconcilable_result();
     }
 
+    // 1 for the table and >= 1 for the compaction groups that overlap
+    utils::small_vector<gate::holder, 2> async_gate_holders;
+    async_gate_holders.push_back(_async_gate.hold());
+    {
+        const auto token_range = range.transform(std::mem_fn(&dht::ring_position::token));
+        for (auto& cg : compaction_groups()) {
+            if (cg.token_range().overlaps(token_range, dht::token_comparator())) {
+                async_gate_holders.push_back(cg.async_gate().hold());
+            }
```

Comment:
There could be quite a few compaction_groups (number of tablets, times 3 when splitting)
So it might be better to define a function to lookup the intersecting compaction groups given a token range, as this can be done efficiently by looking up only the range of storage groups matching the first and last tokens in the range, respectively.

Reply:
Fixed in v2.